tripy/tripy/utils/ast.py

```diff
@@ -130,12 +130,13 @@ def index_into_expr(node: ast.expr, index: int) -> ast.expr:
                 return node
 
             # If we have multiple dimensions specified, then we have a tuple of slices.
-            # Indices are given in as a list of start, stop, step
+            # NOTE: We subtract num_positional from the index because the slice arguments would
+            # be passed as *variadic arguments* to slice_helper and so would come after the positional argument
             if isinstance(node.slice, ast.Tuple):
-                element = node.slice.elts[index // 3]
-                arg_node = index_into_expr(element, index % 3)
+                element = node.slice.elts[(index - num_positional) // 3]
```

Comment:
Why was it working before without the offset?

Reply:
I changed it in two places. We were subtracting one from index to make up for the positional arg when we called `index_into_expr` previously, but I think it makes more sense to include the positional arg in the function call and adjust inside `index_into_expr` instead.